SecureAuth.Sdk/NumberProfileService.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using System.Linq;
+
+namespace SecureAuth.Sdk
+{
+    public class NumberProfileService : INumberProfileService
+    {
+        private readonly ApiClient _apiClient;
+
+        protected internal NumberProfileService(ApiClient apiClient)
+        {
+            this._apiClient = apiClient;
+        }
+
+        /// <summary>
+        /// Evaluate the provided phone number for carrier status and details.
+        /// </summary>
+        /// <param name="userId">The unique user ID.</param>
+        /// <param name="phoneNumber">The phone number for evaluation.</param>
+        /// <returns>NumberProfileResponse</returns>
+        public NumberProfileResponse EvaluatePhoneNumber(string userId, string phoneNumber)
+        {
+            if (string.IsNullOrEmpty(userId))
+            {
+                throw new ArgumentNullException("userId", "User ID cannot be empty.");
+            }
+            if (string.IsNullOrEmpty(phoneNumber))
+            {
+                throw new ArgumentNullException("phoneNumber", "Phone Number cannot be empty.");
+            }
+
+            NumberProfileRequest request = new NumberProfileRequest()
+            {
+                UserId = userId,
+                PhoneNumber = phoneNumber
+            };
+
+            return this._apiClient.Post<NumberProfileResponse>("/api/v1/numberprofile", request);
+        }
+
+        /// <summary>
+        /// Evaluate the provided phone number for carrier status and details.
+        /// </summary>
+        /// <param name="request">NumberProfileRequest object.</param>
+        /// <returns>NumberProfileResponse</returns>
+        public NumberProfileResponse EvaluatePhoneNumber(NumberProfileRequest request)
+        {
+            if (string.IsNullOrEmpty(request.UserId))
+            {
+                throw new ArgumentNullException("userId", "User ID cannot be empty.");
+            }
+            if (string.IsNullOrEmpty(request.PhoneNumber))
+            {
+                throw new ArgumentNullException("phoneNumber", "Phone Number cannot be empty.");
+            }
+
+            return this._apiClient.Post<NumberProfileResponse>("/api/v1/numberprofile", request);
+        }
+
+        /// <summary>
+        /// Updates the original carrier information stored for the user and phone number
+        /// with the updated current carrier information. Used once the user validates
+        /// ownership of the phone number after the carrier change.
+        /// </summary>
+        /// <param name="userId">The unique user ID</param>
+        /// <param name="phoneNumber">The phone number if the record to be updated</param>
+        /// <param name="carrier">Current carrier information from the EvaluatePhoneNumber function.</param>
+        /// <returns>BaseResponse</returns>
+        public BaseResponse UpdateOriginalCarrier(string userId, string phoneNumber, CurrentCarrier carrier)
```

Comment:
I think this is my golang leaking, just a design decision to add a helper method that makes it a bit easier for the implementer, if they didn't want to create the NumberProfileRequest object. The NumberProfileRequest is used both in the update and get functions, so it could be confusing.